endpoints/s3/src/completemultipartupload.cpp

```diff
@@ -416,11 +455,17 @@ void irods::s3::actions::handle_completemultipartupload(
 		std::remove(part_info_vector[i].part_filename.c_str());
 	}
 
+	// clean up shmem
```

Comment:
I thought more about this and I don't think we want to clean up either on failures.  A lot of the failures indicate and invalid request and we may want to allow the client to send a valid request.  I moved these two sections down further so that they only run on success cases.  I also added comments to both indicating why we only want these cleaned up on successes.